game/src/main/scala/io/aigar/game/GameThread.scala

```diff
@@ -70,14 +77,23 @@ class GameThread(scoreThread: ScoreThread) extends Runnable
   }
 
   def transferActions: Unit = {
-    while (!actionQueue.isEmpty) {
-      val action = actionQueue.take
+    var actions = new java.util.ArrayList[ActionQueryWithId]()
+    actionQueue.drainTo(actions)
```

Comment:
I read a little bit about `drainTo` but could you explain me quickly what it is for?